source_code/main_mcu/src/NODEMGMT/nodemgmt.c

```diff
@@ -905,21 +874,20 @@ void nodemgmt_set_data_change_number(uint32_t changeNumber)
  */
 void nodemgmt_set_favorite(uint16_t categoryId, uint16_t favId, uint16_t parentAddress, uint16_t childAddress)
 {
-    nodemgmt_userprofile_t* const dirty_address_finding_trick = (nodemgmt_userprofile_t*)0;
     favorite_addr_t favorite = {parentAddress, childAddress};
     
-    if (categoryId >= (sizeof(dirty_address_finding_trick->category_favorites)/sizeof(dirty_address_finding_trick->category_favorites[0])))
+    if (categoryId >= (MEMBER_SIZE(nodemgmt_userprofile_t, category_favorites)/MEMBER_SIZE(nodemgmt_userprofile_t, category_favorites[0])))
     {
         while(1);
     }
     
-    if(favId >= (sizeof(dirty_address_finding_trick->category_favorites[0].favorite)/sizeof(dirty_address_finding_trick->category_favorites[0].favorite[0])))
+    if(favId >= (MEMBER_SIZE(nodemgmt_userprofile_t, category_favorites[0].favorite)/MEMBER_SIZE(nodemgmt_userprofile_t, category_favorites[0].favorite[0])))
```

Comment:
MEMBER_SUB_ARRAY_SIZE if I'm not mistaken :)

Reply:
Are you sure? I understand that we can fold to MEMBER_ARRAY_SIZE like this:

`if(favId >= (MEMBER_ARRAY_SIZE(nodemgmt_userprofile_t, category_favorites[0].favorite)))`

but, MEMBER_SUB_ARRAY_SIZE(x,y) is equivalent to MEMBER_ARRAY_SIZE(x, y[0]), and we don't have the [0] suffix here

Reply:
you're right!